File: Android/SMK/GoParty/GoPartyAdmin/app/src/main/java/com/example/gopartyadmin/ui/login/LoginActivity.java
package com.example.gopartyadmin.ui.login;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.example.gopartyadmin.ui.MainActivity;
import com.example.gopartyadmin.R;
import com.example.gopartyadmin.koneksi.Api;
import com.example.gopartyadmin.koneksi.ApiConfig;
import com.example.gopartyadmin.model.ModelPemilik;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        SharedPreferences sh = getSharedPreferences("AdminApp", MODE_PRIVATE);
        int id = sh.getInt("id", 0);
        if (id != 0) {
            Intent intent = new Intent(LoginActivity.this, MainActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            startActivity(intent);
        }
    }

    public void regis(View view) {
        Intent intent = new Intent(LoginActivity.this, RegisterActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivity(intent);
    }

    public void login(View view) {
        EditText edtUsername = (EditText) findViewById(R.id.edtUsernameLogin);
        EditText edtPassword = (EditText) findViewById(R.id.edtPasswordLogin);

        String username = edtUsername.getText().toString();
        String password = edtPassword.getText().toString();

        if (username.equals("") || password.equals("")) {
            Toast.makeText(LoginActivity.this, "Isi dengan benar", Toast.LENGTH_SHORT).show();
        } else {
            ApiConfig service = Api.getRetrofit().create(ApiConfig.class);
            Call<List<ModelPemilik>> call = service.getByName(username);
            call.enqueue(new Callback<List<ModelPemilik>>() {
                @Override
                public void onResponse(@NonNull Call<List<ModelPemilik>> call, @NonNull Response<List<ModelPemilik>> response) {
                    if (response.body().isEmpty()) {
                        Toast.makeText(LoginActivity.this, "Akun tidak ada", Toast.LENGTH_SHORT).show();
                    } else if (response.body().get(0).getPassword().equals(password)) {
                        SharedPreferences sharedPreferences = getSharedPreferences("AdminApp", MODE_PRIVATE);
                        SharedPreferences.Editor prefEdit = sharedPreferences.edit();

                        prefEdit.putInt("id", response.body().get(0).getIdpemilik());
                        prefEdit.putString("username", response.body().get(0).getUsername());
                        prefEdit.apply();

                        Intent intent = new Intent(LoginActivity.this, MainActivity.class);
                        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
                        startActivity(intent);
                    } else {
                        Toast.makeText(LoginActivity.this, "Password Salah", Toast.LENGTH_SHORT).show();
                    }
                }

                @Override
                public void onFailure(@NonNull Call<List<ModelPemilik>> call, @NonNull Throwable t) {
                    Toast.makeText(LoginActivity.this, t.getMessage(), Toast.LENGTH_SHORT).show();
                }
            });
        }
    }
}

File: Android/Skripsi/Efeeder/app/src/main/java/com/alcorp/efeeder/ui/LoginActivity.kt
package com.alcorp.efeeder.ui

import android.content.Intent
import android.content.SharedPreferences
import android.os.Build
import android.os.Bundle
import android.view.View
import android.view.WindowInsets
import android.view.WindowManager
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.alcorp.efeeder.databinding.ActivityLoginBinding
import com.alcorp.efeeder.utils.LoadingDialog
import com.alcorp.efeeder.utils.setTimeFormat
import com.alcorp.efeeder.viewmodel.MainViewModel
import com.alcorp.efeeder.viewmodel.ViewModelFactory
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.ktx.auth
import com.google.firebase.ktx.Firebase
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import java.text.SimpleDateFormat
import java.util.*

class LoginActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var binding: ActivityLoginBinding
    private lateinit var auth: FirebaseAuth
    private lateinit var pref: SharedPreferences
    private lateinit var prefEdit: SharedPreferences.Editor

    private val mainViewModel: MainViewModel by viewModels {
        ViewModelFactory.getInstance()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)
        window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN)

        setupToolbar()
        init()
        checkLogin()
    }

    private fun setupToolbar() {
        @Suppress("DEPRECATION")
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window.insetsController?.hide(WindowInsets.Type.statusBars())
        } else {
            window.setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN
            )
        }
        supportActionBar?.hide()
    }

    private fun init() {
        auth = Firebase.auth

        pref = getSharedPreferences("efeeder", MODE_PRIVATE)

        val calendar = Calendar.getInstance()

        val getMonth = SimpleDateFormat("MMMM")
        val getDay = SimpleDateFormat("EEEE")
        val d = Date()

        val dayName = getDay.format(d)
        val month = getMonth.format(calendar.time)

        val year = calendar[Calendar.YEAR]
        val day = calendar[Calendar.DAY_OF_WEEK]

        binding.tvTanggal.text = "$dayName $day $month $year"

        lifecycleScope.launch {
            while(true) {
                mainViewModel.getHour().observe(this@LoginActivity) { hour ->
                    mainViewModel.getMinute().observe(this@LoginActivity) { minute ->
                        binding.tvJam.text = setTimeFormat(hour, minute)
                    }
                }
                delay(1000)
            }
        }

        binding.btnLogin.setOnClickListener(this)
        binding.btnRegis.setOnClickListener(this)
        binding.btnReset.setOnClickListener(this)
        binding.btnLoginTamu.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        when (view) {
            binding.btnRegis -> {
                val intent = Intent(this, RegisActivity::class.java)
                startActivity(intent)
                finish()
            }

            binding.btnReset -> {
                val intent = Intent(this, ResetPasswordActivity::class.java)
                startActivity(intent)
                finish()
            }

            binding.btnLogin -> {
                val loadingDialog = LoadingDialog(this)
                loadingDialog.showDialog()

                val username = binding.edtUsernameLogin.text.toString()
                val password = binding.edtPasswordLogin.text.toString()

                if (username == "" || password == "") {
                    loadingDialog.hideDialog()
                    Toast.makeText(this, "Kolom tidak boleh kosong", Toast.LENGTH_SHORT).show()
                } else {
                    auth.signInWithEmailAndPassword(username, password)
                        .addOnCompleteListener(this) { task ->
                            if (task.isSuccessful) {
                                loadingDialog.hideDialog()
                                auth.currentUser
                                val intent = Intent(this@LoginActivity, MainActivity::class.java)
                                startActivity(intent)
                                finish()
                                Toast.makeText(this@LoginActivity, "Berhasil login", Toast.LENGTH_SHORT).show()
                            } else {
                                loadingDialog.hideDialog()
                                task.addOnFailureListener {
                                    Toast.makeText(baseContext, it.message, Toast.LENGTH_SHORT).show()
                                }
                            }
                        }
                }
            }

            binding.btnLoginTamu -> {
                prefEdit = pref.edit()
                prefEdit.putString("tamu", "tamu")
                prefEdit.apply()

                val intent = Intent(this@LoginActivity, MainActivity::class.java)
                startActivity(intent)
                finish()
                Toast.makeText(this@LoginActivity, "Berhasil login sebagai tamu", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun checkLogin() {
        val tamu = pref.getString("tamu", "")

        val currentUser = auth.currentUser
        if (tamu != null && tamu != "")  {
            val i = Intent(this@LoginActivity, MainActivity::class.java)
            startActivity(i)
            finish()
        }  else if (currentUser != null) {
            val i = Intent(this@LoginActivity, MainActivity::class.java)
            startActivity(i)
            finish()
        }
    }
}